esmvalcore/cmor/_fixes/shared.py

```diff
@@ -8,6 +9,170 @@
 logger = logging.getLogger(__name__)
 
 
+class AtmosphereSigmaFactory(iris.aux_factory.AuxCoordFactory):
+    """Defines an atmosphere sigma coordinate factory."""
+
+    def __init__(self, pressure_at_top=None, sigma=None,
+                 surface_air_pressure=None):
+        """Create class instance.
+
+        Creates and atmosphere sigma coordinate factory with the formula:
+
+        p(n, k, j, i) = pressure_at_top + sigma(k) *
+                        (surface_air_pressure(n, j, i) - pressure_at_top)
+
+        """
+        super().__init__()
+        self._check_dependencies(pressure_at_top, sigma, surface_air_pressure)
+        self.pressure_at_top = pressure_at_top
+        self.sigma = sigma
+        self.surface_air_pressure = surface_air_pressure
+        self.standard_name = 'air_pressure'
+        self.attributes = {}
+
+    @property
+    def units(self):
+        """Units."""
+        units = self.pressure_at_top.units
+        return units
+
+    @staticmethod
+    def _check_dependencies(pressure_at_top, sigma, surface_air_pressure):
+        """Check for sufficient coordinates."""
+        if any([
+                pressure_at_top is None,
+                sigma is None,
+                surface_air_pressure is None,
+        ]):
+            raise ValueError(
+                "Unable to contruct atmosphere sigma coordinate factory due "
+                "to insufficient source coordinates")
+
+        # Check dimensions
+        if pressure_at_top.shape not in ((), (1, )):
```

Comment:
account for empty array ie `(0, )` too?

Reply:
Copy-pasted from another `iris.AuxFactory` which uses a scalar, so I just leave it as it is.